### PR TITLE
[bugfix]  Do not add binding line in both side in reshape map tool

### PR DESCRIPTION
**Pull request by pblottiere**
## Description

When the reshape map tool is used to extend a line whereas the ending point belongs to another line, then both lines are extended. This PR fixes this issue raised here:  https://github.com/qwat/QWAT/issues/221#issuecomment-338694705

Some tests have been added.

## Checklist

- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by pblottiere** [comment]
@nyalldawson Can you give me your opinion on this PR if you have some time? I would like a review before merging this PR :)

File: tests/src/app/testqgsmaptoolreshape.cpp [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  QgsCompoundCurve curve0( *cl0.toCurveType() );
+
+  QgsMapToolReshape tool0( mQgisApp->mapCanvas() );
+  tool0.mCaptureCurve = curve0;
+
+  vl->startEditing();
+  tool0.reshape( vl.get() );
+
+  f0 = vl->getFeature( 1 );
+  QCOMPARE( f0.geometry().asJson(), QStringLiteral( "{\"type\": \"LineString\", \"coordinates\": [ [0, 0], [1, 1], [1, 2], [2, 1]]}" ) );
```
Comment (nyalldawson) [inline tests/src/app/testqgsmaptoolreshape.cpp]: I'd suggest using wkt to compare instead - it's a bit less fragile as there's less variability in the formatting, and is what we generally do in the test

**Comment by nyalldawson** [comment]
I can't fully understand the intricacies of the changes here, but the code looks good and I think the logic is sound, so +1 to merge from me!

**Comment by pblottiere** [comment]
Thank you for your review @nyalldawson!

I'll update unit tests to use WKT for comparing as you suggested.

**Pull request merged**